src/Rector/Latte/RenameMethodLatteRector.php

```diff
@@ -0,0 +1,137 @@
+     */
+    public function configure(array $configuration): void
+    {
+        $methodCallRenames = $configuration[self::RENAME_METHODS] ?? [];
+        Assert::allIsInstanceOf($methodCallRenames, MethodCallRename::class);
+        $this->methodCallRenames = $methodCallRenames;
+    }
+
+    public function changeContent(string $content): string
+    {
+        $typesToVariables = $this->findTypesForVariables($content);
+
+        $methodCallRenames = array_merge(
+            $this->methodCallRenameCollector->getMethodCallRenames(),
+            $this->methodCallRenames
+        );
+        foreach ($methodCallRenames as $methodCallRename) {
+            $oldObjectType = $methodCallRename->getOldObjectType();
+            $className = $oldObjectType->getClassName();
+            $oldMethodName = $methodCallRename->getOldMethod();
+            $newMethodName = $methodCallRename->getNewMethod();
+
+            foreach ($typesToVariables[$className] ?? [] as $classVariableName) {
+                $methodCallPattern = '#\$' . $classVariableName . '->' . $oldMethodName . '\(#';
+                if (Strings::match($content, $methodCallPattern)) {
+                    $content = str_replace(
+                        $classVariableName . '->' . $oldMethodName . '(',
+                        $classVariableName . '->' . $newMethodName . '(',
+                        $content
+                    );
+                }
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * @return array<string, string[]> list of types with all variables of this type
+     */
+    private function findTypesForVariables(string $content): array
+    {
+        $typesToVariables = [];
+
+        $templateTypePattern = '#{templateType (?P<template>.*?)}#';
+        $templateTypeMatch = Strings::match($content, $templateTypePattern);
+        $templateType = $templateTypeMatch['template'] ?? null;
+
+        if ($templateType) {
+            $reflectionClass = ReflectionClass::createFromName($templateType);
+            foreach ($reflectionClass->getProperties() as $property) {
+                /** @var ReflectionNamedType $type */
+                $type = $property->getType();
+
+                if (! isset($typesToVariables[$type->getName()])) {
+                    $typesToVariables[$type->getName()] = [];
+                }
+                $typesToVariables[$type->getName()][] = $property->getName();
+            }
+        }
+
+        $varTypePattern = '#{varType (?P<class>.*?) \$(?P<variable>.*?)}#';
+        $varTypeMatches = Strings::matchAll($content, $varTypePattern);
+
+        foreach ($varTypeMatches as $varTypeMatch) {
+            $className = (string) $varTypeMatch['class'];
+            if (! isset($typesToVariables[$className])) {
+                $typesToVariables[$className] = [];
+            }
+            $typesToVariables[$className][] = $varTypeMatch['variable'];
+        }
+
+        return $typesToVariables;
+    }
```

Comment:
I have troubles reading this code. 
Could you extract it to own service and cover with simple test?

Reply:
extracted